core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -74,6 +74,8 @@ class SYCLInternal {
   size_type* m_scratchFlags = nullptr;
 
   std::optional<sycl::queue> m_queue;
+  static std::vector<std::optional<sycl::queue>*> all_queues;
+  static std::mutex mutex;
```

Comment:
Comment that the mutex is needed for thread safety when inserting or removing elements to `all_queues`